Generated java tests fails when a field has unique validation, entity has filter and is required

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

Same as #9666 maybe is a regression error

Generated java tests fails when a field has unique validation and filter:

```
[ERROR] Tests run: 31, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.886 s <<< FAILURE! - in com.test.unique.web.rest.IssueResourceIT
[ERROR] com.test.unique.web.rest.IssueResourceIT.getAllIssuesByPeopleIsEqualToSomething  Time elapsed: 0.016 s  <<< ERROR!
javax.persistence.PersistenceException: org.hibernate.exception.ConstraintViolationException: could not execute statement
        at com.test.unique.web.rest.IssueResourceIT.getAllIssuesByPeopleIsEqualToSomething(IssueResourceIT.java:385)
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
        at com.test.unique.web.rest.IssueResourceIT.getAllIssuesByPeopleIsEqualToSomething(IssueResourceIT.java:385)
Caused by: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException:
```

##### **Reproduce the error**
Use the JDL from the issue to generate a new project.

```bash
jhipster JDL model.jdl
./mvnw verify
```

##### **Related issues**

#9666

##### **JHipster Version(s)**

7.0.1

##### **JHipster configuration**
 <details>
  <summary>JDL definitions</summary>
  <pre>
     application {
  config {
    applicationType monolith
    authenticationType jwt
    baseName uniquetest
    blueprints []
    buildTool maven
    cacheProvider ehcache
    clientFramework angularX
    clientPackageManager npm
    clientTheme litera
    clientThemeVariant light
    creationTimestamp 1616404743781
    databaseType sql
    devDatabaseType h2Disk
    dtoSuffix DTO
    enableHibernateCache true
    enableSwaggerCodegen false
    enableTranslation true
    jhiPrefix jhi
    jhipsterVersion "7.0.1"
    languages [es]
    messageBroker false
    nativeLanguage es
    otherModules []
    packageName com.test.unique
    prodDatabaseType mysql
    reactive false
    searchEngine false
    serverPort 8080
    serviceDiscoveryType false
    skipClient false
    skipServer false
    skipUserManagement false
    testFrameworks []
    websocket false
    withAdminUi true
  }

  entities Employer, Issue
}

entity Employer {
  email String required unique
  name String required
  telefono1 String required
  telefono2 String
}
entity Issue {
  status String required
  description String required
}

relationship ManyToOne {
  Issue{people(name) required} to Employer
}

service Issue with serviceClass
filter Issue
  </pre>
  </details>

- [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs
+++ b/generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs
@@ -1154,7 +1154,18 @@ _%>
     <%_ } else { _%>
         // Initialize the database
         <%= entityInstance %>Repository.saveAndFlush(<%= persistInstance %>);
-        <%= asEntity(relationship.otherEntityNameCapitalized) %> <%= relationship.relationshipFieldName %> = <%= relationship.otherEntityNameCapitalized %>ResourceIT.createEntity(em);
+    <%_ if (databaseTypeSql && !reactive) { _%>
+        <%= asEntity(relationship.otherEntityNameCapitalized) %> <%= relationship.relationshipFieldName %>;
+    if (TestUtil.findAll(em, <%= asEntity(relationship.otherEntityNameCapitalized) %>.class).isEmpty()) {
+        <%= relationship.relationshipFieldName %> = <%= createEntityPrefix %><%= relationship.otherEntityNameCapitalized %>ResourceIT.createEntity(em);
+        em.persist(<%= relationship.relationshipFieldName %>);
+        em.flush();
+    } else {
+        <%= relationship.relationshipFieldName %> = TestUtil.findAll(em, <%= asEntity(relationship.otherEntityNameCapitalized) %>.class).get(0);
+    }
+    <%_ } else { _%>
+        <%= asEntity(relationship.otherEntityNameCapitalized) %> <%= relationship.relationshipFieldName %> = <%= relationship.otherEntityNameCapitalized %>ResourceIT.createEntity(em);        
+    <%_ } _%>
         em.persist(<%= relationship.relationshipFieldName %>);
         em.flush();
       <%_ if (relationship.relationshipManyToMany || relationship.relationshipOneToMany) { _%>

